td/views.py

```diff
@@ -56,85 +63,141 @@ def get_todo_list_items(request):
+    mongo_db = client[mongo_creds["db_name"]]
+    items_collection = mongo_db.Items
+    reply = items_collection.find({"owner_id": user_int_id},
                                   {"item": 1, "_id": 0})
     items = [document["item"] for document in reply]
+    if len(items) == 0:
+        logger.debug("Items for this user don't exist, reply with "
+                     "{'items': null} JSON.")
+        return {"items": None}
     logger.debug("Found existing items in the database, reply with "
-                 "{'items': %s} JSON object.",
-                 items)
+                 "{'items': %s} JSON object.", items)
     return {"items": items}
 
 
 def add_todo_list_item(request):
     """ Add new item to the database.
 
     When request with POST method at /api/add_todo_list_item arrives then
-    function checks if user's ip exists in the Postgres database table 'Users'.
-    If no then it creates new entry for him in 'Users' Postgres table and after
-    that appends to the 'Items' collection his todo list item from POST json
-    body in Mongo database.
+    function checks if user's id exists in the Postgres database table 'Users'.
+    If no then raises HTTPUnauthorized.
 
     :param request: instance-object which represents HTTP request.
-    :type: pyramid.request.Request
+    :type request: pyramid.request.Request
     :returns: Response instance with 'OK' str body to indicate a success.
     :rtype: pyramid.response.Response
 
     """
     settings = request.registry.settings
     ip = request.client_addr
-    logger.debug("User with ip %s is trying to add new item with POST "
-                 "request on /api/add_todo_list_items",
-                 ip)
-
-    db = settings.db
-    user_id = db.select_one("id", "Users", "ip='%s'" % ip)
-    if user_id is None:
-        logger.debug("User with such ip does not exist in the database, "
-                     "creating new entry for him in Users table")
-        db.insert("Users", "ip", ip)
-        user_id = db.select_one("id", "Users", "ip='%s'" % ip)[0]
-    # Can't use .get here because of errror: TypeError:
-    # 'builtin_function_or_method' object has no attribute '__getitem__'
-    mongo_creds = settings.mongo_creds
+    login_name = authenticated_userid(request)
+    logger.debug("User with ip %s and username '%s' is trying to add new item"
+                 "with POST request on /api/add_todo_list_items",
+                 ip, login_name)
+
+    db = settings["db"]
+    user_int_id = db.select_one("id", "Users", "username='%s'" % login_name)
+    if user_int_id is None:
+        return HTTPUnauthorized()
+    # extract id integer from tuple like (2,)[0] -> 2
+    user_int_id = user_int_id[0]
+    mongo_creds = settings["mongo_creds"]
     client = pymongo.MongoClient(mongo_creds['host'],
                                  int(mongo_creds['port']))
-    db = client[mongo_creds['db_name']]
-    items_collection = db.Items
+    mongo_db = client[mongo_creds['db_name']]
```

Comment:
if client doesn't exists it raises error. Catch it or use `.get(...)` and check `is not None`